src/components/LuxInputAsyncSelect.vue

```diff
@@ -0,0 +1,114 @@
+<template>
+  <div class="autocomplete-container">
+    <lux-autocomplete-input
+      :items="allCurrentItems"
+      @selected="setSelected($event)"
+      @input="findNewItems($event)"
+      ref="autocomplete"
+      :placeholder="placeholder"
+      :label="label"
+      :hide-label="hideLabel"
+      :isAsync="isAsync"
+    />
+    <lux-icon-base class="search-icon"><lux-icon-search></lux-icon-search></lux-icon-base>
+  </div>
+  <!--
+  @slot hiddenInput -- You can use this to pass the user's selected value back to the backend (e.g. Rails) on form submit
+      @binding {string} selectedItem an array of items (objects) that the user has selected
+  -->
+  <slot name="hiddenInput" :selectedItem="selectedItem"></slot>
+</template>
+<script setup>
+import LuxAutocompleteInput from "./LuxAutocompleteInput.vue"
+import LuxIconBase from "./icons/LuxIconBase.vue"
+import LuxIconSearch from "./icons/LuxIconSearch.vue"
+import { computed, ref, useTemplateRef } from "vue"
+
+defineOptions({ name: "LuxInputAsyncSelect" })
```

Comment:
I think it would be nice to include some documentation in the styleguide for when we should use LuxInputAsyncSelect vs. LuxInputSelect vs. LuxAutocompleteInput.  But that could come in a future PR too.  (Putting this comment here, because if I remember right, if you add a jsdoc-style comment above this `defineOptions`, it will end up as a description for the entire component in the styleguide).